### DISCUSSION TITLE
How to implement wouter Link component in polymorphic component?

### DISCUSSION THREAD

**Discussion by ahmadxgani** [mantinedev/mantine#4416] [Q&A] [answered]
i try to add wouter Link as component:
```tsx
{data.map(({ label, link, icon: Icon }, i) =>
  i === 0 ? (
    <UnstyledButton component={Link} href={link} className={classes.control} key={label}>
      <Group position="apart" spacing={0}>
        <Box sx={{ display: "flex", alignItems: "center" }}>
          <ThemeIcon variant="light" size={30}>
            <Icon size="1.1rem" />
          </ThemeIcon>
          <Box ml="md">{label}</Box>
        </Box>
      </Group>
    </UnstyledButton>
  ) : (
    <UnstyledButton<"a"> component="a" href={link} className={classes.control} key={label}>
      <Group position="apart" spacing={0}>
        <Box sx={{ display: "flex", alignItems: "center" }}>
          <ThemeIcon variant="light" size={30}>
            <Icon size="1.1rem" />
          </ThemeIcon>
          <Box ml="md">{label}</Box>
        </Box>
      </Group>
    </UnstyledButton>
  )
)}
```

but it seems the className not applied for wouter Link component but working as well in  builtin HTML tag, Am I missing something?
here's the [full code](https://pastebin.com/8su8dJte)

![image](https://github.com/mantinedev/mantine/assets/73988956/ba14a589-dda9-43de-8803-a4fb5d5183c4)
![image](https://github.com/mantinedev/mantine/assets/73988956/4851c56f-b11a-4ca6-85d2-a23e8bbfb96f)
![image](https://github.com/mantinedev/mantine/assets/73988956/4b045edc-c6a4-43b4-bd22-4cc18dfb7106)
![image](https://github.com/mantinedev/mantine/assets/73988956/00068584-a2fc-4d96-b33d-07d59e81f675)

I apologize for not being able to reproduce the code, so i upload image instead.

**Comment by rtivital** [comment, marked as answer]
Class name might not be applying if the component that is passed to the `component` prop does not support `className` prop.

**Reply by ahmadxgani** [reply]
thanks for the response!
it's possible to convert className to style props? something like [this](https://mantine.dev/core/transition/)
but I can't find the way on [this page](https://mantine.dev/styles/create-styles/)

**Reply by ahmadxgani** [reply]
solved with [this way](https://github.com/molefrog/wouter/issues/247)
thanks for your hint!

**Reply by ahmadxgani** [reply]
but I see there is an oddity that I can't understand, this is just my curiosity, the `<Text component={Link} className={classes.link} href='/'>` works fine, does that mean that the `UnstyledButton` component not passing the prop correctly?

also i try to wrap `UnstyledButton` to my custom component, so it's look like:
```tsx
function LinkButton({
  setOpened,
  opened,
  hasLinks,
  label,
  classes,
  icon: Icon,
}: {
  setOpened: React.Dispatch<React.SetStateAction<boolean>>;
  hasLinks: boolean;
  opened: boolean;
  icon: React.FC<any>;
  label: string;
  classes: { chevron: string; control: string };
}) {
  return (
    <UnstyledButton onClick={() => setOpened((o) => !o)} className={classes.control}>
      <Group position="apart" spacing={0}>
        <Box sx={{ display: "flex", alignItems: "center" }}>
          <ThemeIcon variant="light" size={30}>
            <Icon size="1.1rem" />
          </ThemeIcon>
          <Box ml="md">{label}</Box>
        </Box>
        {hasLinks && (
          <IconChevronRight
            className={classes.chevron}
            size="1rem"
            stroke={1.5}
            style={{
              transform: opened ? `rotate(90deg)` : "none",
            }}
          />
        )}
      </Group>
    </UnstyledButton>
  );
}
```
```tsx
<Link href={link}>
  <LinkButton {...props} /> // but now this not work
</Link>
```
and I'm confused whether this is a problem caused from mantine or wouter
